android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainScreen.kt

```diff
@@ -107,81 +100,85 @@ private fun AppMainNavigationGraph(
   navController: NavHostController,
   mainNavigationScreens: List<MainNavigationScreen>,
   openDrawer: (Boolean) -> Unit,
-  sideMenuOptions: List<SideMenuOption>,
+  navigationConfiguration: NavigationConfiguration,
   measureReportViewModel: MeasureReportViewModel = hiltViewModel(),
   appMainViewModel: AppMainViewModel
 ) {
+  val firstNavigationMenu = navigationConfiguration.clientRegisters.first()
+
+  // registerId = (id of the register's click action) otherwise use navigation menu id
```

Comment:
is this comment explaining the line below? If so, the equal sign is kinda confusing, I'd change to something like this
```suggestion
  // the id of the register's click action, if it exists, otherwise the first navigation menu id
```

Also, this function seems important to explain, I think we should add a full long javadoc comment above `AppMainNavigationGraph`. Maybe this comment could move up there, but not necessarily